src/Http/Handlers/AuthenticateHandler.php

```diff
@@ -32,7 +32,8 @@ public function handle( Request $request, Response $response ): Response {
 			auth_redirect();
 		}
 
-		if ( ! $this->consent_storage->needs_consent( get_current_user_id() ) ) {
+		$client_id = $request->query['client_id'];
```

Comment:
It might be better to do as follows so that it works with both GET and POST requests (not sure if this endpoint can be called by POST though):

```suggestion
		$client_id = $request->request['client_id'];
```

This how we're doing it in `AuthorizeHandler`.

Reply:
I think its always a GET request to authenticate endpoint since user has to be presented with the consent screen. From there, it can be both GET & POST for authorize endpoint as per OIDC spec and we are doing it via POST.

The $request object actually doesn't have anything in its `request` property with a GET request, so I am thinking it doesn't work how PHP treats `$_REQUEST`.

![Screenshot 2022-09-29 at 3 03 27 PM](https://user-images.githubusercontent.com/858906/193015133-ac548ce1-0fca-4581-92d2-682cfcf44750.png)